pkg/skaffold/util/env_template.go

```diff
@@ -131,3 +135,27 @@ func MapToFlag(m map[string]*string, flag string) ([]string, error) {
 
 	return kvFlags, nil
 }
+
+// defaultFunc is a template function that behaves as sprig's default function.
+// See https://masterminds.github.io/sprig/defaults.html#default
+func defaultFunc(dflt, value interface{}) interface{} {
+	if value == nil {
+		return dflt
+	}
+	v := reflect.ValueOf(value)
+	switch v.Kind() {
+	case reflect.Array, reflect.Slice:
```

Comment:
are there other types we want to consider here? or another way of posing this question: how much of https://github.com/Masterminds/sprig/blob/master/defaults.go#L35-L60 should we reuse here?

Reply:
Given that most of our uses will be environment strings, I don't expect any.